crucible-llvm/src/Lang/Crucible/LLVM/Translation/Instruction.hs

```diff
@@ -1820,13 +1820,32 @@ callFunctionWithCont :: forall s arch ret a.
    LLVMGenerator s arch ret a {- ^ continuation for next instructions -} ->
    LLVMGenerator s arch ret a
 callFunctionWithCont instr tailCall_ fnTy fn args assign_f k
+
+     | L.ValSymbol "llvm.dbg.declare" <- fn =
+       do (val, lv, di) <- dbgArgs args
+          ptr <- case asScalar val of
+                   Scalar _ PtrRepr ptr -> pure ptr
+                   _ -> fail "dbg.declare: expected pointer"
+          extensionStmt (LLVM_Debug (LLVM_Dbg_Declare ptr lv di)) >> k
+
+     | L.ValSymbol "llvm.dbg.addr" <- fn =
+       do (val, lv, di) <- dbgArgs args
+          ptr <- case asScalar val of
+                   Scalar _ PtrRepr ptr -> pure ptr
+                   _ -> fail "dbg.addr: expected pointer"
+          extensionStmt (LLVM_Debug (LLVM_Dbg_Addr ptr lv di)) >> k
+
+     | L.ValSymbol "llvm.dbg.value" <- fn =
+       do (val, lv, di) <- dbgArgs args
+          case asScalar val of
+            Scalar _ repr v1 -> extensionStmt (LLVM_Debug (LLVM_Dbg_Value repr v1 lv di)) >> k
+            NotScalar -> fail "dbg.value: expected scalar"
```

Comment:
In these `fail` cases, is it possible to report more about what was actually received? It might speed up debugging when the LLVM folks give us some surprises

Reply:
I'll look to see if there's some code that succinctly describes what LLVM AST we got .

Reply:
Just showing the offending `ScalarView` (it doesn't have a Show instance right now, but looks like it could trivially have one) would probably do the trick.